test/fw/ptl/utils/pbs_anonutils.py

```diff
@@ -529,31 +621,42 @@ def anonymize_file(self, filename, extension=".anon", inplace=False):
 
         f = open(filename)
         for data in f:
-            for k in self.attr_key:
-                if k in data:
-                    if k not in self.gmap_attr_key:
-                        ak = self.utils.random_str(len(k))
-                        self.gmap_attr_key[k] = ak
-                    else:
-                        k = self.gmap_attr_key[k]
-                    data = data.replace(k, self.gmap_attr_key[k])
-
-            for k in self.resc_key:
-                if k not in self.gmap_resc_key:
-                    rk = self.utils.random_str(len(k))
-                    self.gmap_resc_key[k] = rk
-                else:
-                    rk = self.gmap_resc_key[k]
-                data = data.replace(k, self.gmap_resc_key[k])
-
-            for ak, av in self.gmap_attr_val.items():
-                for k, v in av.items():
-                    data = data.replace(k, v)
+            # Get attribute-value pairs from the data
+            kvl = self.__get_kvl(data)
+            if kvl is None:
+                continue
 
-            for ak, av in self.gmap_resc_val.items():
-                for k, v in av.items():
-                    data = data.replace(k, v)
-            nf.write(data)
+            delete_data = False
+            for key, value in kvl.iteritems():
+                if key in self.attr_delete.keys():
+                    if len(kvl.keys()) == 1:
+                        # If there's only one key-value pair in this line
+                        # and this key-value pair should be deleted
+                        # then delete the whole line
+                        delete_data = True
+                    else:
+                        # Delete the key-value pair only
+                        kv_start = data.find(key)
+                        value_index = data.find(value)
+                        kv_end = value_index + len(value)
+                        data = data[:kv_start] + data[kv_end:]
```

Comment:
My math is probably off, but won't this include the first character of the key (and maybe the last character of the value)